kin-devplatform-sdk/src/main/java/kin/devplatform/accountmanager/AccountManagerImpl.java

```diff
@@ -237,7 +245,59 @@ public void onError(Exception e) {
 	public void switchAccount(final int accountIndex, @NonNull final KinCallback<Boolean> callback,
 		@NonNull final IMigrationManagerCallbacks migrationManagerCallbacks) {
 		Logger.log(new Log().withTag(TAG).put("switchAccount", "start"));
-		startMigration(accountIndex, callback, migrationManagerCallbacks);
+		RestorableWalletApi restorableWalletApi = new RestorableWalletApi();
+		try {
+			handleRestorable(accountIndex, callback, migrationManagerCallbacks, restorableWalletApi);
+		} catch (ApiException e) {
+			Logger.log(new Log().priority(Log.ERROR).withTag(TAG).text("getIsRestorableWallet: error is " + e));
+			callback.onFailure(ErrorUtil.createWWalletWasNotCreatedInThisAppException());
+		}
+	}
+
+	//TODO maybe need a better method name...
+	private void handleRestorable(final int accountIndex, @NonNull final KinCallback<Boolean> callback,
+		@NonNull final IMigrationManagerCallbacks migrationManagerCallbacks,
+		RestorableWalletApi restorableWalletApi) throws ApiException {
+		restorableWalletApi.getIsRestorableWallet(blockchainSource.getPublicAddress(accountIndex),
+			new ApiCallback<RestorableWalletRequest>() {
+				@Override
+				public void onFailure(ApiException e, int statusCode,
+					Map<String, List<String>> responseHeaders) {
+					handler.post(new Runnable() {
+						@Override
+						public void run() {
+							Logger.log(
+								new Log().priority(Log.ERROR).withTag(TAG).text("getIsRestorableWallet: onFailure"));
+							callback.onFailure(ErrorUtil.createWWalletWasNotCreatedInThisAppException());
+						}
+					});
+
+				}
+
+				@Override
+				public void onSuccess(final RestorableWalletRequest response, final int statusCode,
+					final Map<String, List<String>> responseHeaders) {
+					handler.post(new Runnable() {
+						@Override
+						public void run() {
+							if (response == null) {
+								Logger.log(new Log().priority(Log.ERROR).withTag(TAG)
+									.text("getIsRestorableWallet: onSuccess but response is null"));
+								callback.onFailure(ErrorUtil.createWWalletWasNotCreatedInThisAppException());
```

Comment:
do we have some general error in this case? "wallet was not created in this app" is not the real cause here

Reply:
I was the only one in all the API classes who dealt with response == null, so there was no real error for that.
need to think about it.